### PR TITLE
Fe feat/detailspage query

### PR DESCRIPTION
**Pull request by Elise-Jones**
# Description

-Changes query for details location and remove passing the state prop thorugh Navlink
-Corrected routing paths 

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [ ] Testing
- [X] Refactoring

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Components/FieldGuide/FieldGuide.js [hunk 1]
```diff
@@ -16,8 +16,9 @@ const FieldGuide = () => {
   console.log(data)
   const renderSavedMushroomCards = () => {
     if (loading) return <p>Loading...</p>;
-    // if (!data || !data.mushrooms) return <p>No mushrooms found.</p>;
-  console.log(data.user.savedMushrooms)
+    if (!data || !data.user.savedMushrooms) return <p>No mushrooms found.</p>;
```
Comment (jmartin777) [inline src/Components/FieldGuide/FieldGuide.js]: great user handling here

File: src/queries.js [hunk 2]
```diff
@@ -80,3 +80,38 @@ export const GET_SAVED_USERMUSHROOMS = gql`
 
 
 `
+export const GET_MUSHROOM_DETAILS = gql `{
```
Comment (jmartin777) [inline src/queries.js]: so there was a query needed  to access details page in the Field Guide

File: src/queries.js [hunk 3]
```diff
@@ -103,15 +102,15 @@ export const GET_MUSHROOM_DETAILS = gql `{
...
-      }		
+      }
       taxonomies {
-            genus
-            order
-            family
-            phylum
-            kingdom
-          }
+        genus
```
Comment (jmartin777) [inline src/queries.js:107]: great catch

**Review by jmartin777 — APPROVED** [review]
Great Refactor here. Great job with the query call

File: src/Components/App/App.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import React, { useState, useEffect } from 'react';
-import { Route, Routes } from 'react-router-dom'; 
+import { Route, Routes, Navigate } from 'react-router-dom';
```
Comment (andreasorensen) [inline src/Components/App/App.js:2]: Are we still using Navigate? We will clean all the imports up at the end.

File: src/Components/App/App.js [hunk 5]
```diff
@@ -78,10 +78,12 @@ function App() {
...
         {showError && <ErrorPage error={error} />} 
         <Routes>
-          <Route path="/" element={<HomePage error={error}  userImage={userImage} onImageUpload={setUserImage} renderMushroomCards={renderMushroomCards}/>} /> 
-          <Route path="/error" element={<ErrorPage error={error} />} /> 
-          <Route path="/myfieldguide" element={<FieldGuide error={error} renderMushroomCards={renderMushroomCards} />} />
-          <Route path="/details/:id" element={<DetailsPage />} />
+          <Route path="/" element={<HomePage error={error} userImage={userImage} onImageUpload={setUserImage} renderMushroomCards={renderMushroomCards} />} />
+          <Route path="/error" element={<ErrorPage error={error} />} />
+          <Route path="/myfieldguide" element={<FieldGuide error={error} />}>
+            <Route path="details/:id" element={<DetailsPage />} />
```
Comment (andreasorensen) [inline src/Components/App/App.js:84]: I'm so glad we figured this out! Amazing that something so simple `../` can make such a huge difference. What a great learning opportunity!

File: src/Components/DetailsPage/DetailsPage.js [hunk 6]
```diff
@@ -1,69 +1,89 @@
 import React from "react";
 import "./DetailsPage.css";
-import { useLocation } from "react-router-dom";
+import {  useParams, redirect } from "react-router-dom";
```
Comment (andreasorensen) [inline src/Components/DetailsPage/DetailsPage.js:3]: Also don't think we need redirect...

File: src/Components/DetailsPage/DetailsPage.js [hunk 7]
```diff
@@ -1,69 +1,89 @@
...
+
 
 const DetailsPage = () => {
-  const location = useLocation();
-  const mushroom = location.state;
-  const taxonomy = mushroom.taxonomies[0];
-  const characteristics = mushroom.characteristics[0];
-  const lookAlikesValue = handleNoLookAlikes(mushroom.lookAlikes);
-  const { isToggled, toggle } = useToggleSave();
+  let mushroomId = useParams().id
```
Comment (andreasorensen) [inline src/Components/DetailsPage/DetailsPage.js:12]: I love that we ended up using this!

File: src/Components/DetailsPage/DetailsPage.js [hunk 8]
```diff
@@ -1,69 +1,89 @@
...
+    return <div>Error: {error.message}</div>;
+  }
+
+  if (!data) {
+    return <div>Data not available.</div>;
+  }
+
+  const {id, description, edibility, photo, commonName, latinName, psychoactive} = data.mushroom;
+  const { genus, order, kingdom, phylum, family } = data.mushroom.taxonomies[0];
+  const {hymeniumType, stipeCharacter, sporePrintColor, mushroomCapShape, hymeniumAttachment, mushroomEcologicalType} = data.mushroom.characteristics[0];
```
Comment (andreasorensen) [inline src/Components/DetailsPage/DetailsPage.js:32]: LOVE DESTRUCTURING!

File: src/Components/FieldGuide/FieldGuide.js [hunk 9]
```diff
@@ -8,40 +8,37 @@ import { NavLink } from 'react-router-dom';
...
-    // if (!data || !data.mushrooms) return <p>No mushrooms found.</p>;
-  console.log(data.user.savedMushrooms)
+    if (!data || !data.user.savedMushrooms.length) return <p>You have no mushrooms saved.</p>;
+  
     return data.user.savedMushrooms.map((mushroom) => (
-      <div className="mushroom-card-wrapper" key={mushroom.id}>
-        <NavLink to={`details/${mushroom.id}`} className="custom-nav-link" state={mushroom}>
+     
+      <div className="mushroom-card-wrapper" key={mushroom.apiKey}>
+        <NavLink to={`../details/${mushroom.id}`} className="custom-nav-link" >
```
Comment (andreasorensen) [inline src/Components/FieldGuide/FieldGuide.js:24]: yassssss! Great job Elise!

File: src/Components/MushroomCard/MushroomCard.js [hunk 10]
```diff
@@ -2,16 +2,17 @@ import React from "react";
...
 import ToggleSave from '../ToggleSave/ToggleSave';
 import { formatProbability } from "../../utils";
-import { useState } from "react";
+import { useState, useQuery } from "react";
 import { useLocation } from 'react-router-dom'
 
+const MushroomCard = ({ id, image, latinName, commonNames, probability }) => {
 
-const MushroomCard = ({ image, latinName, commonNames, probability }) => {
   const [isFav, setIsFav] = useState(false);
```
Comment (andreasorensen) [inline src/Components/MushroomCard/MushroomCard.js:10]: I don't think we need this variable anymore.

**Review by andreasorensen — APPROVED** [review]
We are so close! Way to go Elise!

**Pull request merged**